### ISSUE TITLE
batch size effectiveness on multi-GPU training

### ISSUE THREAD

**Issue by billhhh** [BVLC/caffe#6004]
I am confuse some concept about the effectiveness of batch size on multi-GPU training. Plz correct it if it's wrong

> i.e. if 
(1) Using a batch size of 64 (literally, in the prototxt), and train on a **single GPU**
(2) Using a batch size of 16 (literally, in the prototxt), and train on **4 GPU**
Both of the **actual batch size on training** of these 2 scenarios is 64, right?(according to https://github.com/BVLC/caffe/blob/master/docs/multigpu.md)

If so, is that means both of the scenarios would come out similar results theoretically?

**Comment by billhhh** [comment]
Hello?

**Comment by jukkaho** [comment]
In short, yes. All computed gradients are combined between all the GPUs at the end of a training iteration and they all will get the same gradient information and that should match single GPU case.

With some special loss functions, their results could change depending on batch sizes (if they would try to compare against all other samples within a single batch) but this is not the case for anything that can be found in Caffe at the moment AFAIK.

**Comment by Noiredd** [comment]
From my understanding (which is definitely not exhaustive) it is just like that. I'll close this in favor of a [duplicate post](https://groups.google.com/forum/#!topic/caffe-users/QcjgPO-EfyQ) on caffe-users, where any further discussion should go.
